expand upon proof of presence (#25)
rahelFain/combined-diem-uses-reqs: draft-ietf-diem-requirements.md

Some use cases may contain confidential or sensitive data, and may require mechanisms to protect such data.
For example, this could be realized with encryption of the general emblem data format that will be part of the architecture or by only serving emblems over channels with access control mechanisms.

## Proof of Presence {#proof-pres}

> Since emblems themselves are unable to directly protect assets against attack, emblems indicating assets are entitled to protections may require a mechanism through which violations of their laws or provisions can be verified forensically. This would be particularly poignant for use cases where emblems can be born or taken down dynamically and could be defined in three different levels, listed from weakest to strongest.

Reviewer: ```suggestion
Since emblems themselves are unable to directly protect assets against attack, emblems indicating assets are entitled to protections may require a mechanism through which violations of their laws or provisions can be verified forensically.
This would be particularly relevant in cases where emblems can be applied and removed dynamically.
These protections are defined in three different levels, listed from weakest to strongest.
```

Level 1 - presence and verifiability: Establishing that an actor or querying party was able to obtain the emblem at the
time of violation. That is forensically demonstrating/proving that the emblem was discoverable and verifiable at the
time of an alleged violation.

Level 2 - presence, verifiability and access: Establishing the emblem’s presence and verifiability and that the
querying party accessed the digital emblem.

> Level 3 -  presence, verifiability access and verification: Demonstrating presence verifiability and access and that the querying party verified the emblem upon accessing it.

Reviewer: @rahelFain do you want to explain that Level 3 would only possible for the querying party (how could I tell if you verified a claim or not?), whereas Levels 1 and 2 could be proven by either party depending on the design.
The querying party showing it accessed the emblem (or that definitively no emblem was present) might be a different requirement than the queried party showing that querying party asked for and received an emblem.

```suggestion
Level 3 -  presence, verifiability access and verification: Demonstrating presence verifiability and access and that the querying party verified the emblem upon accessing it. This level of proof can only be made by the querying party.
```

> Note that Levels 2 and 3 are intended to be mutually exclusive requirements with Undetectable Validation {{undet-validation}}.

Reviewer: Just a small nit that Level 2 could be available for the querying party without contradicting any Undetectable Validation requirement, but having queried party Level 2 would contradict any Undetectable Validation requirement.

Author: I'd be fine with adding a footnote to that effect, if it seems necessary.

querying party accessed the digital emblem.

Level 3 -  presence, verifiability access and verification: Demonstrating presence verifiability and access and that the querying party verified the emblem upon accessing it.

Note that Levels 2 and 3 are intended to be mutually exclusive requirements with Undetectable Validation {{undet-validation}}.
An example from the Diplomatic Pouch use case illustrates the Level 3 Proof of Presence requirement.

The use case for diplomatic pouches involves the following entities:
- Point of Entry Country/Customs Agent(s): Validator
- Origin Country or Accredited Organization: Issuer and Authorizing entity
- Diplomat: Agent of Country or Accredited Organization

As described in more detail in Section {#diplo-pouch}, the 1961 Vienna Convention on Diplomatic Relations {{VIENNACONV}} prohibits the
physical inspection of diplomatic pouches (e.g. via X-ray, opening or tampering with the pouch etc.) by customs agents
of a Point of Entry country.
Thus, a Level 3 Proof of Presence record for a digital emblem of this type would establish whether or not customs agent(s)
in the Point of Entry country adhered to the treaty in processing the diplomatic pouch.

> That is to say that they validated the pouch's digital emblem and nothing else.

Reviewer 2: I can see how level 3 can prove that an agent inspected a digital emblem. But I see two problems in how the requirements interact with the use case. Maybe, both can be addressed.

1. Is it mandatory for bags to be inspected? If yes, then indeed, you can verify that procedure was followed by checking that when a bag crosses a border, its emblem was validated. If checking is not mandatory, there is little the emblem proof of access proves, right?
2. I do not see how proof of access proves that the agent did "nothing else." That would require some sort of seal of the bag, i.e., opening the bag destroys the emblem.

Author: Those are both fair points. My plan is to update the text to effectively give the two answers below.

1. Since the emblem denotes that the contents of the pouch are exempt from the normal point of entry customs inspections and restrictions, it is mandatory for the point of entry customs officials to validate the emblem.  However, in the case that the emblem cannot be validated it may be acceptable to use physical markings that could be validated.  - That is to say that the validation of the digital emblem failing may not give the customs officials permission to open or inspect the pouch. Rather, that validation of a diplomatic pouch's digital or physical emblem succeeding indicates it enjoys the treaty's guaranteed protections.

2. Proof of access does not prove that the agent did "nothing else." 
In all of the discussions I've had around this, my sense has been that mechanisms for guaranteeing "nothing else" was done is beyond the scope of what the emblem itself may be able to guarantee. To that end, the Level 3 validation may need to be part of an attested chain of custody (likely some form of SCITT record set or ledger) and/or accompanied by some physical mechanisms for tamper-proofing the pouch/making it impossible to open the pouch, x-ray it etc. without detection or the generation of forensic evidence.

To that end, the Proof of Presence record of a diplomatic pouch's digital emblem could require the following fields

- Specific point of entry
- Time/Date of arrival at point of entry
- identifier(s) of customs agent(s) validating the pouch

> - Record of the pouch's Digital Emblem Validation signed by the customs agent(s)

Reviewer 2: Editing nit: I would move this text into `#diplo-pouch` and simply forward ref here. It's good to have use cases descriptions be self-contained.

Regulates the trans-boundary movement of hazardous wastes. Use cases are functionally identical to OPCW and IAEA.

## Diplomatic Pouches (1961 Vienna Convention on Diplomatic Relations) {#diplo-pouch}

> Digital emblems can protect diplomatic pouch shipments, diplomatic couriers, and diplomatic envoys, who are protected

Reviewer: ```suggestion
Digital emblems can protect diplomatic pouch shipments, diplomatic couriers, and diplomatic envoys.
All three of these are protected
```

under the 1961 Vienna Convention on Diplomatic Relations {{VIENNACONV}}, which states that they may not be stopped, delayed, or
inspected. This creates the paradox that the validity of their credentials must be evaluated, yet doing so has
historically compromised the very rights that are intended to be signaled. Diplomatic markings have also been
misappropriated as cover for the smuggling of drugs and other contraband. Digital emblems, which can be validated

> instantaneously, at a distance, and without interrupting the subject, solve both of these problems, while streamlining

Reviewer: ```suggestion
instantaneously, at a distance, and without interrupting the subject, address both of these problems, while streamlining
```

and automating customs and immigrations processes.

The use case for diplomatic pouches involves the following entities:
- Point of Entry Country/Customs Agent(s): Validator
- Origin Country or Accredited Organization: Issuer and Authorizing entity

> - Diplomat: Agent of Country or Accredited Organization

Reviewer: What about the pouch itself?

As noted in Section {{proof-pres}}, a Level 3 Proof of presence record could help establish
whether or not customs agent(s) in a Point of Entry country adhered to the treaty in processing the diplomatic pouch.
That is to say that they validated the pouch's digital emblem and can reasonably be assumed to have done little to nothing else.

> To that end, the Proof of Presence record of a diplomatic pouch's digital emblem could require the following fields

Reviewer: ```suggestion
As noted in Section {{proof-pres}}, a Level 3 Proof of presence record could help demonstrate, for the benefit of the customs service, that a customs agent(s) in a Point of Entry country validated the diplomatic pouch.
To that end, the Proof of Presence record of processing a diplomatic pouch's digital emblem could include the following information.
```

- Specific point of entry
- Time/Date of arrival at point of entry
- identifier(s) of customs agent(s) validating the pouch
- A baseline record establishing the Emblem's existence and accessibility (or a pointer thereto)
- Record of the customs agent's attempt to validate the Digital Emblem and its result signed by the customs agent(s)

### Limitations of proof of presence:
As indicated in Section {{proof-pres}}, Level 3 Proof of Presence alone may not provide sufficient assurance that
no tampering with the diplomatic pouch or inspection of its contents has occured.
This is because a proof of presence neither provides a chain of custody nor any mechanisms to detect
tampering should it occur. For this reason, Level 3 validation may need to be part of an attested

> chain of custody and/or accompanied by the use of physical mechanisms for tamper-proofing a physical asset.

Reviewer: ```suggestion
As indicated in Section {{proof-pres}}, Level 3 Proof of Presence alone does not provide proof that no tampering with the diplomatic pouch or inspection of its contents has occurred.
This is because a proof of presence neither provides a chain of custody nor any mechanisms to detect tampering should it occur.
For this reason, Level 3 validation may be used along side or as part of an attested chain of custody and/or accompanied by the use of physical mechanisms for tamper-proofing a physical asset.
Any such chain of custody specification or anti-tampering mechanism is out of the scope of the DIEM WG.
```

historically compromised the very rights that are intended to be signaled. Diplomatic markings have also been
misappropriated as cover for the smuggling of drugs and other contraband. Digital emblems, which can be validated
instantaneously, at a distance, and without interrupting the subject, solve both of these problems, while streamlining
and automating customs and immigrations processes. 

The use case for diplomatic pouches involves the following entities:
- Point of Entry Country/Customs Agent(s): Validator
- Origin Country or Accredited Organization: Issuer and Authorizing entity
- Diplomat: Agent of Country or Accredited Organization

As noted in Section {{proof-pres}}, a Level 3 Proof of presence record could help establish 
whether or not customs agent(s) in a Point of Entry country adhered to the treaty in processing the diplomatic pouch.

> That is to say that they validated the pouch's digital emblem and little to nothing else.

Reviewer 2: Okay, I see that you now list in the limitations that it actually may not be possible to verify that "little or nothing else" was done, but why still list it here then as part of level 3 proof of presence?

Author: I was trying to get at the fact that there was still value in establishing that the a digital emblem for the pouch existed and that the validation occurred even if no further protections or checks are used. This option may be chosen in cases where there was less concern about the risk posed by inspection or where further chain of custody and/or anti-tampering mechanisms were not feasible (e.g. possibly due to the size/weight of the contents, a lack of infrastructure or resources to support them, etc.). 

Would this be addressed by adjusting the phrasing to say the following?:
`That is to say that they validated the pouch's digital emblem and can reasonably be assumed to have done little to nothing else.`

To that end, the Proof of Presence record of a diplomatic pouch's digital emblem could require the following fields

- Specific point of entry
- Time/Date of arrival at point of entry
- identifier(s) of customs agent(s) validating the pouch

> - Record of the pouch's Digital Emblem Validation signed by the customs agent(s)

Reviewer 2: What is a proof of presence record?

Author: Based on my understanding of your question, my best answer is that the record is a transcript of the proof... 
Thinking about this I can see that there is a lack of clarity here. Within the confines of this example, I need to distinguish between:
- proving the pouch's digital emblem existed/was accessible at the time of the pouch's arrival 
- proving that customs agents performed the requisite checks of the emblem. 

The Baseline proof of presence would largely be the first thing. 
This would then need to be accompanied by the signed validation attempt and its result to establish the second.

Edit to note that I've made these changes.
